base/lnls-prune-artifacts.sh

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env bash
+
+set -eu
+
+
+filter_out_dirs() {
+    list="$1"
+    exclude_list="$2"
+
+    for item in $exclude_list; do
+        dir=$item
+
+        while [ "$dir" != "/" ]; do
+            list=$(echo "$list" | grep -xv $dir)
+
+            dir=$(dirname $dir)
+        done
+    done
+
+    echo "$list"
+}
+
+find_elf_executables() {
+    targets=$@
+
+    while read -r executable; do
+        read -r -N 4 trailer < "$executable"
+
+        # Output only ELF binaries
+        if [ "$trailer" = $'\x7fELF' ]; then
+            echo $executable
+        fi
+    done < <(find $targets -type f -executable)
+}
+
+find_linked_libraries() {
+    executables=$(find_elf_executables $@)
+
+    # Depend on the glibc-specific behavior of supporting multiple executables
+    # to be queried at once
+    linked=$(ldd $executables 2>/dev/null | grep '=>')
+
+    # We grep out not found libraries, since they cannot be kept if we don't
+    # know where they are.
+    #
+    # Final binary may be actually runnable, since rpath of another binary may
+    # pull those not found libraries
+    found="$(echo "$linked" | grep -v "not found")"
+
+    # Get their full path
+    libs=$(echo "$found"| cut -d'>' -f 2 | cut -d' ' -f 2)
```

Comment:
I feel like there is a better way to do this. I though abort `grep -o`, but it would require a pattern for any library path, including soname handling. I'm not sure if that turns out to be better.

Reply:
> Why doesn't using one `cut -d' ' -f 3` command work?

Actually, it does. I was trying not to rely on spaces at first, but I turned out doing so in the end. I didn't notice it could be easily simplified. Thanks.